jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
+
+        List<String> values=_preferredEncodingOrderCache.get(key);
+        if (values==null)
+        {
+            QuotedEncodingQualityCSV encodingQualityCSV = new QuotedEncodingQualityCSV(_precompressedFormats);
```

Comment:
The problem passing in the precompressedFormats here is that they still need to be processed.
I think it would be better to have a static Trie of the known precompressed formats that maps to the value.  Then here we just pass in a static function that looks up that Trie.   No garbage created.